chirp/drivers/kguv980p.py

```diff
@@ -2460,28 +2424,55 @@ def _decode(lst):
                            RC_POWER_LIST[_settings.rc_power]))
         remote_grp.append(rs)
 
-        _code = ''
-        test = int(_settings.ani_edit)
-        _codeobj = '0x{0:0{1}X}'.format(test, 6)
-        LOG.debug("codeobj = %s" % _codeobj)
-        _ani = str(_codeobj)
-        for i in range(2, 8):
-            LOG.debug("ani[i] = %s" % _ani[i])
-            if _ani[i] in dtmfchars:
-                _code += _ani[i]
-        val_ani = int(_code)
+        def decode_remote_vals(self, setting):
+            _code = ''
+            # convert the stored 24bit decimal value to 6 individual
+            # digits in hex format into a string.
+            # in the radio an id of 000000 is stored as 0x000000
+            # a id value of 101 is stored as 0x101CCC
+            # pad the value with up to 6 zeroes to handle
+            # default value of 000000 and not simply display 0
+            value = int(setting)
+            _codeobj = "{0:0{1}X}".format(value, 6)
+            LOG.debug("codeobj = %s" % _codeobj)
+            _val = _codeobj
+            # parse each char in _codeobj and only add to
+            # _code if it is a valid DTMF char
+            # for a value of 0x101CCC we only want to display 101
+            for i in range(0, 6):
+                LOG.debug("val [i] = %s" % _val[i])
+                if _val[i] in dtmfchars:
+                    _code += _val[i]
+            val_out = int(_code)
+            return val_out, _code
+
+        def apply_remote_id(setting, obj, val):
+            # convert the 3-6 digit id to the proper hex value
+            # and pad with 0 if length is < 3 (5 with 0x prefix)
+            # if len of id > 3 but < 6 (excl 0x) pad the rest with "C"
+            val2 = hex(int(str(val), 16))
+            if (int(val2, 16) != 0):
+                while len(val2) < 5:
+                    val2 += '0'
+                while len(val2) < 8:
+                    val2 += 'C'
+            value = int(str(val2), 16)
```

Comment:
This stuff like L2453 is just really roundabout. `hex(int(str('101'), 16))` is just `0x101`. The same as val, but with `0x` in front right?

I think this whole function can be replaced with this:
```
val.ljust(5, '0').ljust(8, 'C')
```
Can't it?

Reply:
this appears to work:
        def apply_remote_id(setting, obj, val):
            val = str(val)
            value = int(val.ljust(3, '0').ljust(6, 'C'), 16)
            setattr(obj, setting.get_name(), value)